### PR TITLE
feat: create tabs component with info and data

### PR DESCRIPTION
**Pull request by almog8k**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Further  information:
create tabs component with info and data(implementing react json schema form) pages

Screenshot from 2024-06-03 14-42-57

image

### CODE REVIEW

File: src/components/Tabs/CustomTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const routesMatch = tabs.map((tab) => tab.path);
+
+  const routeMatch = useRouteMatch(routesMatch);
+
+  const currentTab = routeMatch?.pattern?.path;
+  return (
+    <>
+      <Tabs value={currentTab}>
+        {tabs.map((tab) => (
+          <Tab key={tab.path} label={tab.label} value={tab.path} component={Link} to={tab.path} />
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/components/Tabs/CustomTabs.tsx:26]: we cant use it, we tried so hard and failed.

File: src/components/Tabs/CustomTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        ))}
+      </Tabs>
+      <Box sx={{ width: '100%' }}>{currentTab && tabs.find((tab) => tab.path === currentTab)?.component}</Box>
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/components/Tabs/CustomTabs.tsx:29]: done

File: src/components/queryWrapper/queryWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { Box, CircularProgress, Typography } from '@mui/material';
+
+type queryWrapperProps = {
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/components/queryWrapper/queryWrapper.tsx:3]: done

File: src/components/queryWrapper/queryWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  isLoading: boolean;
+  error: Error | null;
+  isSuccess: boolean;
+  children: React.ReactNode;
+};
+
+export const QueryWrapper: React.FC<queryWrapperProps> = ({ isLoading, error, isSuccess, children }) => {
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/components/queryWrapper/queryWrapper.tsx:10]: maybe QueryDataRenderer

File: src/hooks/useQueryParams.ts
[the remark this answers is not in the document]
Comment (almog8k) [reply src/hooks/useQueryParams.ts:1]: file deleted

File: src/pages/config/configInfo/configInfoPage.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  key: string;
+  value: string | number | React.ReactNode;
+};
+
+export const ConfigInfoPage: React.FC<ConfigInfoPageProps> = ({ configInfo }) => {
+  if (!configInfo) {
+    return <div>Config not found</div>;
+  }
+
+  const configTPresent: ListData[] = [
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/config/configInfo/configInfoPage.tsx:20]: yes, hot T

File: src/pages/config/configInfo/configInfoPage.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                      {key}
+                    </Typography>
+                  }
+                  secondary={
+                    <Typography variant="body2" color="textSecondary">
+                      {value}
+                    </Typography>
+                  }
+                />
+              </ListItem>
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/config/configInfo/configInfoPage.tsx:47]: done

File: src/pages/config/configInfo/configInfoPage.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import { Box, List, ListItem, ListItemText, Paper, Typography } from '@mui/material';
+import { config } from '../../../api/client';
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+type ConfigInfoPageProps = {
+  configInfo: config | undefined;
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/config/configInfo/configInfoPage.tsx:7]: no we don't

File: src/pages/configs/configsPage.tsx [hunk 8]
```diff
@@ -1,33 +1,30 @@
...
 export const ConfigsPage: React.FC = () => {
-  const { data, error } = useQuery({ queryKey: ['configs'], queryFn: () => getConfigs() });
+  const { data, error, isLoading, isSuccess } = useQuery({ queryKey: ['configs'], queryFn: () => getConfigs() });
 
-  if (error) {
-    console.error(error);
-  }
-
-  const createText = 'Create Config';
+  const configData = useMemo(() => data?.configs ?? [], [data]);
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/configs/configsPage.tsx:13]: done

File: src/pages/config/configPage.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    if (!data) return [];
+    return [
+      { label: 'Info', path: encodeURI(`${routes.CONFIG}/${name}/${version}`), component: <ConfigInfoPage configInfo={data} /> },
+      {
+        label: 'Data',
+        path: encodeURI(`${routes.CONFIG}/${name}/${version}/data`),
+        component: <ViewConfigPage config={data} />,
+      },
+      {
+        label: 'Json',
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/pages/config/configPage.tsx:33]: changed

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**